lib/js/src/rpc/enums/HMILevel.js

```diff
@@ -0,0 +1,91 @@
+/*
+* Copyright (c) 2019, Livio, Inc.
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* Redistributions of source code must retain the above copyright notice, this
+* list of conditions and the following disclaimer.
+*
+* Redistributions in binary form must reproduce the above copyright notice,
+* this list of conditions and the following
+* disclaimer in the documentation and/or other materials provided with the
+* distribution.
+*
+* Neither the name of the Livio Inc. nor the names of its contributors
+* may be used to endorse or promote products derived from this software
+* without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+* ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+* LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import { Enum } from '../../util/Enum.js';
+
+/**
+ * @typedef {Enum} HMILevel
+ * @property {Object} MAP
+ */
+class HMILevel extends Enum {
+
+    constructor() {
+        super();
+    }
+
+    /**
+     * @return {String} 
+     */
+    static get HMI_FULL() {
```

Comment:
@nickschwab 
```

    <enum name="HMILevel" since="1.0">
        <description>Enumeration that describes current levels of HMI.</description>
        <element name="FULL" internal_name="HMI_FULL" />
        <element name="LIMITED" internal_name="HMI_LIMITED" />
        <element name="BACKGROUND" internal_name="HMI_BACKGROUND" />
        <element name="NONE" internal_name="HMI_NONE" />
    </enum>
```
Based on the XML above what consistensy do you mean? This enum has the `internal_name` attribute for each element. In my opinion, the consistency is to use `internal_name` if that exists. In this case `HMI_` prefix should be present.

Reply:
Yes, if we decide that `internal_name` should be used for keys and getter/setter method names, then all of these should be updated to include the `HMI_` prefix.

Reply:
I really dislike a lot of the internal names because the redundancy created with the prefixes. For example this enum access would be `HMI_STATUS.HMI_FULL` when the `HMI_STATUS` is the context given for the enum value as it should be.  _However_, since that is how it is in the RPC spec and seeing how javascript actually needs the `internal_name` for some cases (ie, `BitRate`) it does seem like the better approach to simply use it versus the `name`.

Reply:
@joeygrover this also allows simplifying the code generation as we shouldn't use a lot of custom transformation rules and just could follow the `internal_name` instead. I'm not sure why `HMI_` prefix was added here, it seems like this enum doesn't require any `internal_name`, but if this needs to be removed, it should be done in XML itself.

Reply:
I don't think it would be safe to change in the spec as it would likely cause some breaking changes somewhere.

Reply:
I agree, but, in my opinion, this is only the correct way because the inconsistency with the prefix arrived from the XML in this case. Moreover, we've analyzed all libraries and found too many mismatches in naming, sometimes even no any ideas from where the name came, eg. Objective-C enums.

Reply:
I understand it appears as only an "xml file" but it is the spec for RPC messages. So removing or changing values is inherently a breaking change. And at this point we do not have a proper insight on how that would affect the other projects. Because we would be changing the spec we would have to update all projects to properly reflect this change as well. It is simply not worth the effort; we work towards building higher level abstraction APIs over the RPCs anyways so developers will be shielded from these missteps.